packages/mui-joy/src/Badge/Badge.tsx

```diff
@@ -101,11 +102,12 @@ const BadgeBadge = styled('span', {
     ownerState.anchorOrigin?.horizontal === 'left' ? 'translateX(-50%)' : 'translateX(50%)';
   const transformOriginY = ownerState.anchorOrigin?.vertical === 'top' ? '0%' : '100%';
   const transformOriginX = ownerState.anchorOrigin?.horizontal === 'left' ? '0%' : '100%';
-  const typography =
-    theme.typography[`body-${({ sm: 'xs', md: 'sm', lg: 'md' } as const)[ownerState.size!]}`];
+  const typography = theme.typography[
+    `body-${({ sm: 'xs', md: 'sm', lg: 'md' } as const)[ownerState.size!]}`
+  ] as CSSObject | undefined;
```

Comment:
I don't think we need to typecast here. We just need to handle undefined case by using optional chaining as done below.
```suggestion
  ];
```